craft_cli/dispatcher.py

```diff
@@ -359,17 +359,24 @@ def pre_parse_args(
             else:
                 filtered_sysargs.append(sysarg)
 
-        # control and use quiet/verbose options
-        if sum(global_args[key] for key in ("quiet", "verbose", "trace")) > 1:
+        # control and use quiet/verbose/verbosity options
+        if sum(1 for key in ("quiet", "verbose", "verbosity") if global_args[key]) > 1:
             raise ArgumentParsingError(
-                "The 'verbose', 'trace' and 'quiet' options are mutually exclusive."
+                "The 'verbose', 'quiet' and 'verbosity' options are mutually exclusive."
             )
         if global_args["quiet"]:
             emit.set_mode(EmitterMode.QUIET)
         elif global_args["verbose"]:
             emit.set_mode(EmitterMode.VERBOSE)
-        elif global_args["trace"]:
-            emit.set_mode(EmitterMode.TRACE)
+        elif global_args["verbosity"]:
+            try:
+                verbosity_level = EmitterMode[global_args["verbosity"].upper()]
+            except KeyError:
+                raise ArgumentParsingError(  # pylint: disable=raise-missing-from
+                    "Bad verbosity level; allowed are "
+                    "'quiet', 'brief', 'verbose', 'debug' and 'trace'."
+                )
+            emit.set_mode(verbosity_level)
```

Comment:
is it correct that set_mode still does not understand  DEBUG and is that expected for this PR?

Reply:
`set_mode` (like `init`) will not crash, they will set `emit`'s level to DEBUG, but the Emitter will not behave yet as we want for this level, that's intended for the next branch.